lib/ansible/modules/windows/win_optional_feature.ps1

```diff
@@ -0,0 +1,108 @@
+#!powershell
+
+# Copyright: (c) 2019, Carson Anderson <rcanderson23@gmail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#AnsibleRequires -CSharpUtil Ansible.Basic
+
+$spec = @{
+	options = @{
+		name = @{ type = "str"; required = $true }
+		state = @{ type = "str"; default = "present"; choices = @("absent", "present") }
+		source = @{ type = "str" }
+		include_parent = @{ type = "bool"; default = $false }
+	}
+	supports_check_mode = $true
+}
+
+$module = [Ansible.Basic.AnsibleModule]::Create($args, $spec)
+$module.Result.rc = 0
```

Comment:
rc should only be used for module that interact with an executable. The `failed` key is the proper one to use and by calling `FailJson()` this will set that automatically for you.